custom/gcc_sangath/tests/test_custom_actions.py

```diff
@@ -0,0 +1,151 @@
+from datetime import date
+from unittest.mock import patch
+
+from casexml.apps.case.mock import CaseFactory, CaseIndex, CaseStructure
+
+from corehq.apps.data_interfaces.models import AutomaticUpdateRule
+from corehq.apps.data_interfaces.tests.test_auto_case_updates import (
+    BaseCaseRuleTest,
+)
+from corehq.apps.data_interfaces.tests.util import create_empty_rule
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.users.models import CommCareUser
+from corehq.apps.users.util import normalize_username
+from corehq.form_processor.models import CommCareCase
+from custom.gcc_sangath.const import (
+    DATE_OF_PEER_REVIEW_CASE_PROP,
+    MEAN_GENERAL_SKILLS_SCORE_CASE_PROP,
+    MEAN_TREATMENT_SPECIFIC_SCORE_CASE_PROP,
+    PEER_RATING_CASE_TYPE,
+    SESSION_CASE_TYPE,
+    SESSION_RATING_CASE_PROP,
+)
+from custom.gcc_sangath.rules.custom_actions import (
+    sanitize_session_peer_rating,
+)
+
+
+class SanitizeSessionPeerRatingTest(BaseCaseRuleTest):
+    domain = 'gcc-sangath'
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.project = create_domain(cls.domain)
```

Comment:
Just confirming, do the domain and user both need to exist? It doesn't look like `create_empty_rule` requires a real domain, but I didn't look into `CaseFactory`.

Reply:
I think i can get away with user. We are replicating a real case update later, so a missing domain might create an issue, will check and remove if possible.

Reply:
https://github.com/dimagi/commcare-hq/pull/31119/commits/0209374f28232d432b2003ac9b290bd6bd747f85